backend/src/test/java/ai/verta/modeldb/TestsInit.java

```diff
@@ -165,15 +166,44 @@ public static void setServerAndService() throws Exception {
   }
 
   @AfterClass
-  public static void removeServerAndService() {
+  public static void removeServerAndService() throws InterruptedException {
     App.initiateShutdown(0);
 
+    cleanUpResources();
+
+    // shutdown test server
+    serverBuilder.build().shutdownNow();
+  }
+
+  protected static void cleanUpResources() throws InterruptedException {
     // Remove all entities
     // removeEntities();
     // Delete entities by cron job
-    deleteEntitiesCron.run();
+    SoftDeleteProjects softDeleteProjects = ReconcilerInitializer.softDeleteProjects;
+    SoftDeleteExperiments softDeleteExperiments = ReconcilerInitializer.softDeleteExperiments;
+    SoftDeleteExperimentRuns softDeleteExperimentRuns =
+        ReconcilerInitializer.softDeleteExperimentRuns;
+
+    softDeleteProjects.resync();
+    while (softDeleteProjects.isElementsNotDeleted()) {
+      LOGGER.trace("Project deletion is still in progress");
+    }
+    softDeleteExperiments.resync();
+    while (softDeleteExperiments.isElementsNotDeleted()) {
+      LOGGER.trace("Experiment deletion is still in progress");
+    }
+    softDeleteExperimentRuns.resync();
+    while (softDeleteExperimentRuns.isElementsNotDeleted()) {
+      LOGGER.trace("ExperimentRun deletion is still in progress");
+    }
 
-    // shutdown test server
-    serverBuilder.build().shutdownNow();
+    ReconcilerInitializer.softDeleteDatasets.resync();
+    ReconcilerInitializer.softDeleteRepositories.resync();
```

Comment:
It would be fine if we had a resync cronjob here too, right? Not sure why we have to have that `if` on the testing condition.

Reply:
that if added due to https://github.com/VertaAI/modeldb/pull/2708#discussion_r756090704